geolocation/success.https.html

```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<meta charset="utf-8"/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js?feature=bidi"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+
+<script>
+    promise_setup(async () => {
+        // Ensure permission is granted before proceeding.
+        await test_driver.bidi.permissions.set_permission({
+            descriptor: {name: "geolocation"},
+            state: "granted",
+        });
+
+        // Ensure any previously set geolocation emulations are cleared.
+        await test_driver.bidi.emulation.set_geolocation_override(
+            {coordinates: null});
+    });
+
+    promise_test(async (t) => {
+        const mockLatitude = 51.478;
+        const mockLongitude = -0.166;
+        const mockAccuracy = 100;
+
+        let position;
+
+        await test_driver.bidi.emulation.set_geolocation_override(
```

Comment:
This seems potentially racy. it might be better to hold promises for `navigator.geolocation.getCurrentPosition((position)` and `test_driver.bidi.emulation.set_geolocation_override`, and then await each, and then check the position result from navigator.geolocation.getCurrentPosition(), like:

```
const setPositionPromise = test_driver.bidi.emulation.set_geolocation_override(...);
const getPositionPromise = new Promise((resolve, reject) => navigator.geolocation.getCurrentPosition(resolve, reject);
await setPositionPromise;
try {
   const position = await getPositionPromise;
    // assert test things here 
} catch (err) {
   assert_unreached
}
const position 
```

Reply:
Agree. Fixed.